dynamics/client/base.py

```diff
@@ -92,6 +92,7 @@ def __init__(
         resource: Optional[str] = None,
         *,
         cache_token: bool = True,
+        connection_timeout: int = 5,
```

Comment:
Type should be Optional[int], since you can disable timeout with None, as incidated by the docstring.